setup.py

```diff
@@ -102,7 +102,7 @@ def build_extension(self, ext):
         "jaxlib>0.4.1",
         "networkx",
         "numpy",
-        "pymbar>3.0.4,<4",
+        "pymbar>=3.0.6,<4",
```

Comment:
nit: Should this be 3.1.0 or higher? Not sure why this is different from requirements

Reply:
`<3.0.6` definitely won't work because of https://github.com/choderalab/pymbar/pull/425, which was merged in `3.0.6`. In general I prefer to use version constraints in `setup.py` only to exclude known-incompatible versions.

When upgrading, I opted to use the latest release <4.